core/src/commonMain/kotlin/com/xebia/functional/xef/vectorstores/LocalVectorStore.kt

```diff
@@ -46,9 +46,20 @@ private constructor(private val embeddings: Embeddings, private val state: Atomi
     }
   }
 
-  override suspend fun memories(conversationId: ConversationId, limit: Int): List<Memory> {
+  override suspend fun memories(conversationId: ConversationId, limitTokens: Int): List<Memory> {
     val memories = state.get().orderedMemories[conversationId]
-    return memories?.takeLast(limit).orEmpty().sortedBy { it.timestamp }
+    return memories
+      ?.foldRight(Pair(0, emptyList<Memory>())) { memory, (accTokens, list) ->
```

Comment:
This seems to be similar in all implementations, can we do it in a single place and call the function?